.github/workflows/test.yml

```diff
@@ -1,323 +1,85 @@
-# TODO the next great simplification might be deploying multiple examples to one dfx replica instance: https://forum.dfinity.org/t/use-the-same-local-replica-for-multiple-projects/11900
-# TODO this might allow us to avoid spinning up so many jobs in the matrix
-
 # This GitHub Action flow works as follows:
-# The tests are currently simple example-based integration tests.
-# Each directory in the examples directory represents an example project and is intended to have tests that ensure the canisters contained in that example function properly.
+# Each directory in the examples and tests directory represents an example project and is intended to have tests that ensure the canisters contained in that example function properly.
 # These tests are currently written in TypeScript and are intended to be run in a Node.js environment.
 # This GitHub Action takes care of deploying to npm and GitHub.
-# Tests can either run against the current code of Azle found in the repository, or the code deployed by the GitHub Action to npm.
-# Feature branch pull requests (pull requests without release-- in the base branch name) will run all tests against the code found in the repository.
-# Release branch pull requests (pull requests with release-- in the base branch name) will run all tests against the code found in the repository and the code deployed by the GitHub Action to npm.
-# Pushes to main will run all tests against the code in the repository if the latest commit was not a merge of a release branch, and will run tests against the code in the repository and the code deployed by the GitHub Action to npm otherwise.
-# The basic-integration-tests matrix spins up one job per combination of example directory and code source (repo or npm).
-# The check-basic-integration-tests-success job is designed to ensure that all jobs spun up from the matrix in the basic-integration-tests have succeeded
 
-name: Azle Tests
+# TODO give every step a name
+
+name: Tests
 on:
     push:
         branches:
             - main
     pull_request: # Runs on pull requests to any branch
+env:
+    IS_RELEASE_CANDIDATE: ${{ contains(github.head_ref, 'release--') }}
```

Comment:
Rename variables as discussed in other comments

Reply:
done